bin/hipcc

```diff
@@ -116,20 +117,26 @@ if ($HIP_PLATFORM eq "hcc") {
 
         # Only include the libstdc++ headers and libraries flags explicitly if the g++ is older than version 5.
         # That's because HCC already uses libstdc++ by default if a newer g++/libstdc++ is available
-        if (${GCC_CUR_VER} eq ${GPP_CUR_VER} and $GPP_VER_FIELDS[0] < 5) {
+        # Cent OS 7 cannot use libstdc++ for compilation, defaults to libc++
+        if (${GCC_CUR_VER} eq ${GPP_CUR_VER} and $GPP_VER_FIELDS[0] < 5 and ($HOST_OSNAME ne "\"centos\"")) {
             $HCC_WA_FLAGS .= " -stdlib=libstdc++  -I/usr/include/x86_64-linux-gnu -I/usr/include/x86_64-linux-gnu/c++/${GCC_CUR_VER} -I/usr/include/c++/${GCC_CUR_VER} ";
             # Add C++ libs for GCC.
             $HIPLDFLAGS .= " -lstdc++";
         }
     }
 
     # Force -stdlib=libc++ on UB14.04
-    $HOST_OSNAME= `cat /etc/os-release | grep "^ID\=" | cut -d= -f2 | tr -d '\n'`;
     $HOST_OSVER= `cat /etc/os-release | grep "^VERSION_ID\=" | cut -d= -f2 | tr -d '\n'`;
     if ($HOST_OSNAME eq "ubuntu" and $HOST_OSVER eq "\"14.04\"") {
         $HIPCXXFLAGS .= " -stdlib=libc++";
         $setStdLib = 1;
     }
+    # Force -stdlib=libc++ on Cent OS 7
+    if ($HOST_OSNAME eq "\"centos\"" and $HOST_OSVER eq "\"7\"") {
+        $HIPCXXFLAGS .= " -stdlib=libc++";
+        $HIPLDFLAGS .= " -lc++ -lc++abi";
+        $setStdLib = 1;
```

Comment:
hipcc calls hcc-config to get the appropriate c++ compilation flags.  We already sets to libc++ on centos/rhel in hcc-config:
https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/hcc_config/hcc_config.cpp#L103

If there's some flags missing we should add them to hcc-config instead of hipcc